packages/react/src/checkbox/Checkbox.tsx

```diff
@@ -34,24 +43,45 @@ export type CheckboxProps = BaseCheckboxProps & {
 
 export const Checkbox = forwardRef<HTMLInputElement, CheckboxProps>(
 	function Checkbox(
-		{ children, disabled, invalid: invalidProp, size = 'md', ...props },
-		ref
+		{
+			children,
+			disabled,
+			invalid: invalidProp,
+			size = 'md',
+			indeterminate,
+			...props
+		},
+		forwardedRef
 	) {
+		const ref = useRef<HTMLInputElement>(null);
 		const controlGroupContext = useControlGroupContext();
 		const invalid = invalidProp || controlGroupContext?.invalid;
+
+		// `indeterminate` cannot be set using an HTML attribute
```

Comment:
> This is set using the [HTMLInputElement](https://developer.mozilla.org/en-US/docs/Web/API/HTMLInputElement) object's indeterminate property via JavaScript (it cannot be set using an HTML attribute):

Read more about this here https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox#indeterminate_state_checkboxes

Reply:
Might be good to put this comment in code too? (with the link)